### PR TITLE
zap only devices we care about

### PR DESCRIPTION
**Pull request by Lassulus**

### CODE REVIEW

**Comment by Lassulus** [comment]
uhm I guess I killed garnix

**Comment by Mic92** [comment]
> uhm I guess I killed garnix

Just flaky webhooks. More often than enough it's github's fault.

File: disk-deactivate [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#!/bin/sh
+# dependencies: jq util-linux lvm2 mdadm zfs
+disk=$1
+
+lsblk --output-all --json | jq -r --arg disk_to_clear "$disk" '
```
Comment (Mic92) [inline disk-deactivate:5]: I think at this lenght, the jq script could become its own file. Not sure if editors have syntax highlighting for that....

Comment (Lassulus) [reply disk-deactivate:5]: well it mostly is it's own file? or should we move the jq code to another extra file?

Comment (Mic92) [reply disk-deactivate:5]: Yep. I mean the jq script.

**Comment by Mic92** [comment]
The ci error looks legitimate.

**Comment by Lassulus** [comment]
> The ci error looks legitimate.

looks like nondeterminism for me.

File: disk-deactivate/disk-deactivate.jq [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      "vgchange -a n \"$vg\"",
+      "vgremove -f \"$vg\""
+    ]
+  elif .fstype == "swap" then
+    "swapoff \(.path)"
+  elif .fstype == null then
+    # maybe its zfs
+    [
+      # the next line has some horrible escaping
+      "zpool=$(zdb -l \(.path) | sed -nr $'s/ +name: \\'(.*)\\'/\\\\1/p')",
```
Comment (Mic92) [inline disk-deactivate/disk-deactivate.jq:17]: What argument are you passing to `zdb` here? Is this the device name?

Comment (Lassulus) [reply disk-deactivate/disk-deactivate.jq:17]: the device path, for example '/dev/sda3' or '/dev/mapper/crypted'. we do this to find out if the device is part of a zpool and get the zpool name then. a typical zdb -l output looks like this:
```
------------------------------------
LABEL 0
------------------------------------
    version: 5000
    name: 'rpool'
    state: 0
    txg: 26685541
    pool_guid: 6968731913350947701
    errata: 0
    hostid: 579054254
    hostname: 'prism'
    top_guid: 185435801962358030
    guid: 8148307730435269824
    vdev_children: 1
    vdev_tree:
        type: 'mirror'
        id: 0
        guid: 185435801962358030
        metaslab_array: 256
        metaslab_shift: 31
        ashift: 9
        asize: 254980915200
        is_log: 0
        create_txg: 4
        children[0]:
            type: 'disk'
            id: 0
            guid: 11507650001572448411
            path: '/dev/disk/by-id/ata-Micron_1100_MTFDDAK256TBN_173718FA4342-part3'
            whole_disk: 0
            DTL: 643
            create_txg: 4
        children[1]:
            type: 'disk'
            id: 1
            guid: 8148307730435269824
            path: '/dev/disk/by-id/ata-Micron_1100_MTFDDAK256TBN_170515ACA5C0-part3'
            whole_disk: 0
            DTL: 133
            create_txg: 4
    features_for_read:
        com.delphix:hole_birth
        com.delphix:embedded_data
    labels = 0 1 2 3
```

Comment (Mic92) [reply disk-deactivate/disk-deactivate.jq:17]: Ok. Also don't know about a better way just now.

Comment (Lassulus) [reply disk-deactivate/disk-deactivate.jq:17]: yeah, sadly lsblk has bad zfs support. thats also on of the reason why blkdeactivate what not enough

**Pull request merged**